numerics/matrix_computations.hpp

```diff
@@ -0,0 +1,114 @@
+﻿
+#pragma once
+
+#include "quantities/named_quantities.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_arrays {
+
+using quantities::Quotient;
+using quantities::SquareRoot;
+
+template<typename U>
+struct CholeskyGenerator;
+
+template<typename Scalar_, template<typename S> typename UpperTriangularMatrix>
```

Comment:
Move the definition of the generators to the _body so that this file remains easy to skim.